hxd/fs/FileEntry.hx

```diff
@@ -1,34 +1,142 @@
 package hxd.fs;
 
-class FileEntry {
+/**
+	The base class representing a single file in the FileSystem.
 
+	Should not be instantiated directly, as specific implementation lies on the FileSystem type used.
+**/
+class FileEntry {
+	/**
+		The name of the file.
+	**/
 	public var name(default, null) : String;
+	/**
+		The local path of the file.
+	**/
 	public var path(get, never) : String;
+	/**
+		The directory path the file is placed in.
+	**/
 	public var directory(get, never) : String;
+	/**
+		File extension of the file without the dot (i.e. `png`).
+	**/
 	public var extension(get, never) : String;
+	/**
+		The size of the file in bytes.
+	**/
 	public var size(get, never) : Int;
+	/**
+		Whether this entry is a directory.
+
+		Depending on FileSystem used, it may contain directory entries.
+		In that case they can be iterated over, but should not be accessed as files.
+	**/
 	public var isDirectory(get, never) : Bool;
+	/**
+		Whether this entry is loaded and its contents can be accessed. Otherwise `FileEntry.load` have to be called.
+
+		Depending on FileSystem, some files cannot be accessed immediately and have to be loaded first.
+		In that case `isAvailable` is `false` until file is loaded.
+	**/
 	public var isAvailable(get, never) : Bool;
 
-	// first four bytes of the file
+	/**
+		Returns the first four bytes of the file which usually is a file signature.
+	**/
 	public function getSign() : Int return 0;
 
+	/**
+		Returns the contents of the file as Bytes.
+	**/
 	public function getBytes() : haxe.io.Bytes return null;
 
+	/**
+		Returns the contents of the file as a String.
+	**/
 	public function getText() return getBytes().toString();
 
+	/**
+		Prepares file to be read from the start or moves the reading position to the beginning if its already opened.
+
+		This is an alternative way to read file data without storing its entirety in the RAM.
+		Note that depending on FileSystem used, they always load entire file into RAM, such as EmbedFileSystem or BytesFileSystem.

```

Comment:
I don't think specifics should be documented here.

Reply:
Specifics as in examples of which stock FS do that, specifics about sometimes it loading full file in the RAM or specifics about how this entire set of meant to be utilized?

Reply:
Prepares file to be read from the start or moves the reading position to the beginning if its already opened.
This is a way to read data progressively, but some filesystem might still preload some of all of the data into memory.